src/waveform/renderers/waveformmarkset.cpp

```diff
@@ -60,15 +65,27 @@ void WaveformMarkSet::setup(const QString& group, const QDomNode& node,
 
             if (controlItemSet.insert(hotCueControlItem).second) {
                 //qDebug() << "WaveformRenderMark::setup - Automatic mark" << hotCueControlItem;
-                m_marks.push_back(m_defaultMark);
-                WaveformMark& mark = m_marks.back();
-                mark.setKeyAndIndex(pHotcue->getKey(), i);
+                m_marks.push_back(new WaveformMark(m_pDefaultMark.get()));
+                WaveformMark* mark = m_marks.back();
+                mark->setKeyAndIndex(pHotcue->getKey(), i);
             }
         }
     }
 }
 
 void WaveformMarkSet::clear() {
-    m_defaultMark = WaveformMark();
+    m_pDefaultMark = std::make_unique<WaveformMark>();
+    qDeleteAll(m_marks);
     m_marks.clear();
 }
+
+const WaveformMark* WaveformMarkSet::getHotCueMark(int hotCue) {
```

Comment:
You are handing out internal pointers for which you retain ownership. On the other hand such a pointer can easily be invalidated during the lifetime of WaveformMarkSet, just be invoking WaveformMarkSet::clear(). This is very dangerous.

Better store the elements of WaveformMarkSet as shared pointers.